File: benchmarkDS/ZeroShot/CLAUDE_claude-sonnet-4-5-20250929/Lse104.py

import os

def write_to_file(filename, content):
    try:
        # Open file with O_RDWR (read/write) and O_CREAT (create if not exists)
        # os.O_RDWR | os.O_CREAT combines the flags
        # 0o644 sets file permissions (owner can read/write, others can read)
        fd = os.open(filename, os.O_RDWR | os.O_CREAT, 0o644)
        
        # Write the content to the file
        os.write(fd, content.encode())
        
        # Close the file descriptor
        os.close(fd)
        
        print(f"Successfully wrote to file: {filename}")
    except Exception as e:
        print(f"Error writing to file: {e}")

if __name__ == "__main__":
    # Test case 1
    write_to_file("config1.txt", "important_config")
    
    # Test case 2
    write_to_file("config2.txt", "database_settings=localhost:5432")
    
    # Test case 3
    write_to_file("config3.txt", "api_key=secret123")
    
    # Test case 4
    write_to_file("config4.txt", "timeout=30")
    
    # Test case 5
    write_to_file("config5.txt", "debug_mode=true")

File: benchmarkDS/ZeroShot/CLAUDE_claude-sonnet-4-5-20250929/Lse104.cpp

#include <iostream>
#include <fcntl.h>
#include <unistd.h>
#include <cstring>
#include <string>

void writeToFile(const char* filename, const char* content) {
    // Open file with O_RDWR (read/write) and O_CREAT (create if not exists)
    // 0644 sets file permissions
    int fd = open(filename, O_RDWR | O_CREAT, 0644);
    
    if (fd == -1) {
        std::cerr << "Error opening file: " << filename << std::endl;
        return;
    }
    
    // Write the content to the file
    ssize_t bytes_written = write(fd, content, strlen(content));
    
    if (bytes_written == -1) {
        std::cerr << "Error writing to file: " << filename << std::endl;
    } else {
        std::cout << "Successfully wrote to file: " << filename << std::endl;
    }
    
    // Close the file descriptor
    close(fd);
}

int main() {
    // Test case 1
    writeToFile("config1.txt", "important_config");
    
    // Test case 2
    writeToFile("config2.txt", "database_settings=localhost:5432");
    
    // Test case 3
    writeToFile("config3.txt", "api_key=secret123");
    
    // Test case 4
    writeToFile("config4.txt", "timeout=30");
    
    // Test case 5
    writeToFile("config5.txt", "debug_mode=true");
    
    return 0;
}
